components/raftstore/src/store/worker/cleanup_snapshot.rs

```diff
@@ -0,0 +1,176 @@
+                    // peer must have been exist. But now it's disconnected, so the peer
+                    // has to be destroyed instead of being created.
+                    info!(
+                        "region is disconnected, remove snaps";
+                        "region_id" => region_id,
+                        "snaps" => ?snaps,
+                    );
+                    for (key, is_sending) in snaps {
+                        let snap = match self.snap_mgr.get_snapshot_for_gc(&key, is_sending) {
+                            Ok(snap) => snap,
+                            Err(e) => {
+                                error!(%e;
+                                    "failed to load snapshot";
+                                    "snapshot" => ?key,
+                                );
+                                continue;
+                            }
+                        };
+                        self.snap_mgr.delete_snapshot(&key, snap.as_ref(), false);
+                    }
+                    Ok(())
+                }
+                Err(TrySendError::Full(_)) => Ok(()),
+                Err(TrySendError::Disconnected(_)) => unreachable!(),
+            }
+        };
+        for (key, is_sending) in snap_keys {
+            if last_region_id == key.region_id {
+                keys.push((key, is_sending));
+                continue;
+            }
+
+            if !keys.is_empty() {
+                schedule_gc_snap(last_region_id, keys)?;
+                keys = vec![];
+            }
+
+            last_region_id = key.region_id;
+            keys.push((key, is_sending));
+        }
+        if !keys.is_empty() {
+            schedule_gc_snap(last_region_id, keys)?;
+        }
+        Ok(())
+    }
+
+    fn delete_snapshot(&self, key: &SnapKey, snap: &Snapshot, check_entry: bool) -> bool {
+        self.snap_mgr.delete_snapshot(key, snap, check_entry)
+    }
+}
+
+impl<EK, ER> Runnable for Runner<EK, ER>
+where
+    EK: KvEngine,
+    ER: RaftEngine,
+{
+    type Task = Task;
+
+    fn run(&mut self, task: Task) {
+        match task {
+            Task::GcSnapshot => {
+                if let Err(e) = self.handle_snap_mgr_gc() {
+                    error!(?e;
+                        "handle gc snap failed";
+                        "store_id" => self.store_id,
+                    );
+                }
+
+                let msg = StoreMsg::GcSnapshotFinish;
+                if let Err(e) = StoreRouter::send(&self.router, msg) {
+                    error!(%e; "send StoreMsg::GcSnapshotFinish failed");
+                }
+            }
+            Task::DeleteSnapshotFiles {
+                key,
+                mut snapshot,
+                check_entry,
+            } => {
+                let snapshot = snapshot.as_mut();
+                if let Err(e) = snapshot.load_snapshot_meta_if_necessary() {
+                    warn!(
```

Comment:
Why still try to delete it when it fails to load meta?

Reply:
In case the meta file is corrupted, we still need to delete the ssts in the best effort. 
The delete_snapshot() will try to find possible sst candidates files to delete.